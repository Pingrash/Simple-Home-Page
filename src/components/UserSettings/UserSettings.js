import React, { useState } from 'react';
import PropTypes from 'prop-types';

import Typography from '@material-ui/core/Typography';
import FormControl from '@material-ui/core/FormControl';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(theme => ({
  form: {
    display: 'flex',
    flexDirection: 'column',
    paddingTop: '2rem'
  },
  formSpacing: {
    margin: '1rem 0'
  },
  success: {
    color: 'green',
    textAlign: 'center'
  }
}));

const UserSettings = ({ user, updateUser }) => {
  const [username, setUsername] = useState('');
  const [submitted, setSubmitted] = useState(false);
  const [userDeleted, setUserDeleted] = useState(false);

  const classes = useStyles();

  return (
    <>
      <form
        data-test='form'
        className={classes.form}
        onSubmit={e => {
          e.preventDefault();
          updateUser(username);
          setSubmitted(true);
          setUsername('');
        }}
      >
        <Typography component='h4' variant='h4'>
          Update User
        </Typography>
        <FormControl>
          <InputLabel
            htmlFor='username-input'
            color='primary'
            required
          >
            New Username:{' '}
          </InputLabel>
          <Input
            id='username-input'
            type='text'
            onChange={e => {
              setUsername(e.target.value);
              if (submitted || userDeleted) {
                setSubmitted(false);
                setUserDeleted(false);
              }
            }}
            value={username}
            autoComplete='true'
            fullWidth
            className={classes.formSpacing}
            placeholder={user}
          />
          <Button
            type='submit'
            color='primary'
            fullWidth
            variant='contained'
            className={classes.formSpacing}
          >
            Submit
          </Button>
        </FormControl>
      </form>
      {submitted ? (
        <Typography
          className={classes.success}
          component='span'
          variant='h4'
        >
          User Updated
        </Typography>
      ) : null}
    </>
  );
};

UserSettings.propTypes = {
  user: PropTypes.string.isRequired,
  updateUser: PropTypes.func.isRequired
};

export default UserSettings;
