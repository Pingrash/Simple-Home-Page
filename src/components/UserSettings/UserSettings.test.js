import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import {
  findByDataTest,
  checkProps,
  testFunc
} from '../../../Utilities';
import UserSettings from './UserSettings';

const props = {
  user: 'username',
  updateUser: testFunc
};

const setup = () => {
  const component = shallow(<UserSettings {...props} />);
  return component;
};

describe('Snapshot', () => {
  it('Should match snapshot', () => {
    const tree = renderer
      .create(<UserSettings {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('UserSettings renders correctly', () => {
  let wrapper;
  beforeEach(() => (wrapper = setup()));

  it('Should render without errors', () => {
    expect(wrapper.length).toBe(1);
  });

  it('Should render a user change form', () => {
    const form = findByDataTest(wrapper, 'form');
    expect(form.length).toBe(1);
  });
});

describe('UserSettings should have correct props', () => {
  it('Should not return any prop errors', () => {
    const propsError = checkProps(UserSettings, props);
    expect(propsError).toBeUndefined();
  });
});
