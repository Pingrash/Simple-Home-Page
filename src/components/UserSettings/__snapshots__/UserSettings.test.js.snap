// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Should match snapshot 1`] = `
<form
  className="makeStyles-form-1"
  data-test="form"
  onSubmit={[Function]}
>
  <h4
    className="MuiTypography-root MuiTypography-h4"
  >
    Update User
  </h4>
  <div
    className="MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
      data-shrink={false}
      htmlFor="username-input"
    >
      New Username:
       
      <span
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
      >
        â€‰
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-formSpacing-2 MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="true"
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="username-input"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="username"
        required={false}
        type="text"
      />
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formSpacing-2 MuiButton-containedPrimary MuiButton-fullWidth"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Submit
      </span>
    </button>
  </div>
</form>
`;
